src/conditionals/primary-category-conditional.php

```diff
@@ -44,6 +45,6 @@ public function is_met() {
 		 * @param array $admin_pages List of additional admin pages.
 		 */
 		$additional_pages = \apply_filters( 'wpseo_primary_category_admin_pages', [] );
-		return \in_array( $this->current_page->get_current_admin_page(), \array_merge( [ 'edit.php', 'post.php', 'post-new.php' ], $additional_pages ), true );
+		return \in_array( $this->current_page->get_current_admin_page(), \array_merge( [ 'edit.php', 'post.php', 'post-new.php', 'admin-ajax.php' ], $additional_pages ), true );
```

Comment:
I would prefer if we narrowed the surface of our fix, to avoid a big impact.

What I would do is to return true on AJAX calls only if it's the specific AJAX call we're interested in, aka the one that's performed by the link component.

So, basically something like this:
```
		$current_page = $this->current_page->get_current_admin_page();

		if ( $current_page === 'admin-ajax.php' && isset( $_POST['action'] ) && $_POST['action'] === 'wp-link-ajax' ) {
			return true;
		}

		/**
		 * Filter: Adds the possibility to use primary category at additional admin pages.
		 *
		 * @param array $admin_pages List of additional admin pages.
		 */
		$additional_pages = \apply_filters( 'wpseo_primary_category_admin_pages', [] );
		return \in_array( $current_page, \array_merge( [ 'edit.php', 'post.php', 'post-new.php' ], $additional_pages ), true );
```

Reply:
Done!